app/src/main/java/com/example/topmovies/adapter/MoviesAdapter.kt

```diff
@@ -55,26 +56,47 @@ class MovieAdapter(private val onItemClickListener: (String) -> Unit) :
                 textviewItemLayoutRankNumber.text = movie.rank
                 textviewItemLayoutYearNumber.text = movie.year
                 textviewItemLayoutPreviousRankNumber.text = movie.rankUpDown
-                when (movie.rankUpDown.first()) {
-                    '+' -> textviewItemLayoutPreviousRankNumber.setTextColor(
-                        ContextCompat.getColor(itemView.context, R.color.text_rank_up)
-                    )
-                    '-' -> textviewItemLayoutPreviousRankNumber.setTextColor(
-                        ContextCompat.getColor(itemView.context, R.color.text_rank_down)
-                    )
-                }
                 circleAvatarViewItemLayoutMovieImage.setLabel(movie.title)
                 Glide.with(circleAvatarViewItemLayoutMovieImage)
                     .asBitmap()
                     .load(resizeImage(movie.imageUrl))
                     .into(avatarCustomTarget)
+                imageButtonItemFavoriteIcon.setImageResource(movieIsFavorite(movie.isFavorite))
+                setRankUpDownColor(movie.rankUpDown)
+                imageButtonItemFavoriteIcon.setOnClickListener {
```

Comment:
Please group together actions that are contextually bound.
In this case,  please do not interrupt your list of operations being performed on `imageButtonItemFavoriteIcon`